### ISSUE TITLE
Upgrade to keras 2.0. Also, build a better infrastructure for testing models and saving states - e.g. each time a model is tested, the code to create that model, the code to create the features, should all be saved. And features, therefore, should be more modular.

### ISSUE THREAD

**Issue by hillarysanders** [hillarysanders/deep_learning#28]
- [x] upgrade to keras 2.0
- [x] attempt to remove all global vars that can be easily removed
- [x] break out features into more sub-features with better arguments used
- [x] add in features and model code snapshot to history callback
  - [x] actually holding off on this for now since just the 1st level feature function definition is used --> kind of actually want this for now. However, even without improvements to the logging system, if there were a set of pretty stable feature sub-functions that were used, that that would still be good to get going for code stability, clarity.
- [x] figure out if the normalization isn't working as well as it used to due to upgrading to keras 2.0 (ask josh about this)
 - [x] test old keras vs new keras AUC accuracy
       # looks like it's comparable - mayyybe a little bit worse / variable but looks pretty solid.
- [ ] hook up tensorboard to the convent output results, customize a bit as needed
   - [x] 1) read tensorboard README: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tensorboard/README.md
   - [ ] 2) read tensorboard tutorials https://www.tensorflow.org/get_started/summaries_and_tensorboard
      - [x] tutorial 1
      - [x] tutorial 2
     - [x] tutorial 3
- [ ] start using the big URL dataset with convnet; compare against splits_5_5
- [ ] look into html URI encoding: Punycode!

**Comment by hillarysanders** [comment]
- finished up URL db stuff, dupe scan_ids in dynamo vt_url_feed, but db is ready to go (43m, 41m unique)
- moved to keras 2.02, using Josh's normalization. Lol it's nice because I no longer have to spin up a local webserver to serve deprecated docs.
- switched to validation generator, not sure if it'll be good enough since we need a ton (slow!) of data for validation
- right now working on metrics, so that AUC, etc, anything we want is calculated by keras by inserting our own metrics into the model
